api/proto/teleport/legacy/types/types.proto

```diff
@@ -974,6 +974,14 @@ message ProvisionTokenSpecV2 {
   ];
   // GitHub allows the configuration of options specific to the "github" join method.
   ProvisionTokenSpecV2GitHub GitHub = 7 [(gogoproto.jsontag) = "github,omitempty"];
+  // AgentMatcherLabels is a set of labels to be used by agents to match on resources.
+  // When an agent uses this token, the agent should monitor resources that match those labels.
+  // For databases, this means adding the labels to `db_service.resources.labels`.
+  wrappers.LabelValues AgentMatcherLabels = 8 [
```

Comment:
why not use `SuggestedLabels` instead of adding this?

Reply:
SuggestedLabels are labels for the resource itself.
They are used in Server Access, the node itself will receive those labels.

It's the equivalent of passing the `--labels` argument when starting an agent
https://goteleport.com/docs/database-access/guides/postgres-self-hosted/#step-45-start-the-database-service
```bash
$ teleport db start \
   --token=/tmp/token \
   --auth-server=teleport.example.com:3080 \
   --name=test \
   --protocol=postgres \
   --uri=postgres.example.com:5432 \
   --labels=env=dev
```

Or the equivalent of `db_service.resources.labels` when configuring the `teleport.yaml` for the teleport when running as a DB service.
https://goteleport.com/docs/database-access/reference/configuration/#database-service-configuration

```yaml
db_service:
  # Enables the Database Service.
  enabled: "yes"

  # Matchers for database resources created with "tctl create" command.
  resources:
  - labels:
      "*": "*"
```

Reply:
It seems to me that `SuggestedLabels` are used for exactly this purpose for the SSH server install script, but I don't know that much about it, I could be misreading, it's the first time I've seen these. Just wondering if we really need to add another field to the token resource. https://github.com/gravitational/teleport/blob/698e377fb412824d48d908a5f8229bc20586e2ca/lib/web/join_tokens.go#L124-L136

Thanks for the explanation on `AgentMatcherLabels` but I don't understand how it is different in behaviour from `SuggestedLabels`, is there some difference between the labels the db will get vs the db service?

Reply:
> It seems to me that SuggestedLabels are used for exactly this purpose for the SSH server install script

`SuggestedLabels` are used to populate the resource's labels - part of its **metadata** properties.
Every resource has those.
They are specific labels and do not follow the matching rules (things like wildcard matching).

`AgentMatcherLabels` are used to populate agent's resource labels - part of its **configuration**.

So, if a database X has the label `env: prod` and our agent has the matching label:
- `env: stg` - database service/agent does not proxy the database X
- `env: *` - database service/agent proxies the database X
- `'*': '*' - database service/agent proxies the database X


Even though they both a set of Labels, they mean totally different things
I mean, we can re-use the field, it respects the *type* that we require **but** that would be misusing this property

Sorry if this was not properly explained, or if I'm missing something

Reply:
oh sorry I completely misunderstood what these were for then, thanks for explaining, let me take a closer look